benchmarks/multiplatform/benchmarks/src/commonMain/kotlin/benchmarks/lazygrid/LazyGrid.kt

```diff
@@ -26,32 +26,25 @@ import androidx.compose.ui.semantics.contentDescription
 import androidx.compose.ui.semantics.semantics
 import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
+import kotlin.coroutines.suspendCoroutine
 
 @Composable
-fun LazyGrid() {
+fun LazyGrid(smoothScroll: Boolean = false, withLaunchedEffectInItem: Boolean = false) {
     val itemCount = 12000
     val entries = remember {List(itemCount) { Entry("$it") }}
     val state = rememberLazyGridState()
 
-    var smoothScroll by remember { mutableStateOf(false)}
+    var scrollIteration by remember { mutableStateOf(0) }
```

Comment:
Why do we need this extra state?

Reply:
So the `state.scrollToItem(curItem)` won't be cancelled before it completes.

When LaunchedEffect takes `curItem` as a key it will be cancelled after `curItem` value changes. 

https://developer.android.com/develop/ui/compose/side-effects#launchedeffect

> If LaunchedEffect is recomposed with different keys (see the [Restarting Effects](https://developer.android.com/develop/ui/compose/side-effects#restarting-effects) section below), the existing coroutine will be cancelled and the new suspend function will be launched in a new coroutine.

Reply:
As we discussed, we'll use 'Unit' as a key for LaunchedEffect. The effect will run a loop and call the scrolling. This way the LaunchedEffect won't be affected by the changes of `curItem` value.